cmd/root.go

```diff
@@ -0,0 +1,136 @@
+/*
+Copyright © 2019 NAME HERE <EMAIL ADDRESS>
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package cmd
+
+import (
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"strings"
+
+	"github.com/smallfish/simpleyaml"
+	"github.com/spf13/cobra"
+
+	homedir "github.com/mitchellh/go-homedir"
+	"github.com/spf13/viper"
+)
+
+var cfgFile string
+
+// rootCmd represents the base command when called without any subcommands
+var rootCmd = &cobra.Command{
+	Use:   "sygaldry",
+	Short: "Usage: sygaldry <step name> -f <rune yaml file>",
+	Long:  `Usage: sygaldry <step name> -f <rune yaml file>`,
+	Run: func(cmd *cobra.Command, args []string) {
+		file := cmd.Flag("file")
+
+		if !strings.HasSuffix(file.Value.String(), ".yaml") {
+			fmt.Printf("Your runes file should be a yaml.")
+		} else if len(args) == 1 && strings.HasSuffix(file.Value.String(), ".yaml") {
+			parseYaml(args[0], file.Value.String())
+		} else {
+			fmt.Println("Usage: sygaldry <step name> -f <rune yaml file>")
+		}
+	},
+}
+
+// Execute adds all child commands to the root command and sets flags appropriately.
+// This is called by main.main(). It only needs to happen once to the rootCmd.
+func Execute() {
+	if err := rootCmd.Execute(); err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+}
+
+func init() {
+	// Here you will define your flags and configuration settings.
+	// Cobra supports persistent flags, which, if defined here,
+	// will be global for your application.
+	cobra.OnInitialize(initConfig)
+
+	// Here you will define your flags and configuration settings.
+	// Cobra supports persistent flags, which, if defined here,
+	// will be global for your application.
+	rootCmd.PersistentFlags().StringVar(&cfgFile, "config", "", "config file (default is $HOME/.sygaldry.yaml)")
```

Comment:
Do we have plans for this config file?

Reply:
Naw, this came default with the Cobra initialization. Haven't put much thought into it just yet. We can rip it out and reimplement it later when we have a better idea for the use cases.

Reply:
Maybe this is where we can list default rune defs